redis/tests/test_async.rs

```diff
@@ -781,3 +782,40 @@ mod mtls_test {
         }
     }
 }
+
+#[test]
+fn test_set_client_name_by_config() {
+    const CLIENT_NAME: &str = "TEST_CLIENT_NAME";
+    use redis::RedisError;
+    let ctx = TestContext::with_client_name(CLIENT_NAME);
+
+    block_on_all(async move {
+        let mut con = ctx.async_connection().await?;
+
+        let client_info: String = redis::cmd("CLIENT")
+            .arg("INFO")
+            .query_async(&mut con)
+            .await
+            .unwrap();
+
+        let client_attrs: Vec<&str> = client_info.split(' ').collect();
```

Comment:
you're using the same logic, at the very least move the splitting into attribute pairs to a separate function. If you really want to make it handy, have it return `HashMap<String, String>`.

Also, there's no need for the collecting. `for` statements  work on iterators:
```rs
for (key, value) in client_info.split(' ').map(|line| attr.split('=').collect::<(&str, &str)>()) {
```
should be sufficient.